src/test/unit/cpu.test.js

```diff
@@ -38,15 +38,8 @@ describe('processThreadCPUDelta', function () {
     return { profile, thread, processedThread };
   }
 
-  it('removes the threadCPUDelta array if all of its values are null', function () {
-    const { processedThread } = setup([null, null, null, null, null, null]);
-    expect(processedThread.samples.threadCPUDelta).toBe(undefined);
-  });
-
-  it('does not remove the values if there is at least one non-null value ', function () {
-    const { processedThread } = setup([null, null, null, null, 0.1]);
-    // We do the processing for null elements, see the following test for a more specific assertion.
-    expect(processedThread.samples.threadCPUDelta).not.toBe(undefined);
+  it('throws if all of its values are null', function () {
+    expect(() => setup([null, null, null, null, null, null])).toThrow();
```

Comment:
Should we really throw here? Maybe we should fail gracefully. But I guess if we encounter some of these things, it's better to throw so we can see if there is a bug in the code instead of ignoring it.

Reply:
This case shouldn't happen, so that's why I prefer to throw :-)